frontend/views/components/UsersSelector.vue

```diff
@@ -0,0 +1,189 @@
+<template lang='pug'>
+form.c-search-form(@submit.prevent='')
+  label.field
+    .sr-only {{label}}
+    .inputgroup.c-search(
+      @click='clear'
+    )
+      .is-icon.prefix(aria-hidden='true')
+        i.icon-search
+      .input(
+        type='text'
+        name='search'
+      )
+        .profile-wrapper(
+          contenteditable='false'
+          v-for='(username, index) in usernames'
+          :key='index'
+        )
+          .profile
+            avatar-user(:username='username' size='xs')
+            .c-name.has-text-bold {{ displayName(username) }}
+            .button.is-icon-small(
+              @click.prevent.stop='remove(username)'
+              :aria-label='L("Clear search")'
+            )
+              i.icon-times
+        .c-keyword(
+          contenteditable
+          ref='input'
+          data-test='users-selector'
+          @keydown='onHandleKeyDown'
+          @keyup='onHandleKeyUp'
+        )
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import AvatarUser from '@components/AvatarUser.vue'
+import { decode } from 'html-entities'
+
+export default ({
+  name: 'UsersSelector',
+  components: {
+    AvatarUser
+  },
+  props: {
+    label: {
+      type: String,
+      required: true
+    },
+    usernames: {
+      type: Array,
+      default: []
+    },
+    defaultValue: {
+      type: String,
+      default: ''
+    },
+    autofocus: {
+      type: Boolean,
+      default: false
+    }
+  },
+  data () {
+    return {
+      value: ''
+    }
+  },
+  computed: {
+    ...mapGetters(['ourContactProfiles'])
+  },
+  mounted () {
+    this.$refs.input.innerHTML = this.defaultValue
+    this.value = this.defaultValue
+    if (this.autofocus) {
+      this.$refs.input.focus()
+    }
+  },
+  methods: {
+    remove (username) {
+      this.$emit('remove', username)
+    },
+    displayName (username) {
+      return this.ourContactProfiles[username].displayName || username
+    },
+    clear () {
+      this.$refs.input.focus()
+      this.$refs.input.innerHTML = ''
+      this.value = ''
+    },
+    onHandleKeyDown (e: KeyboardEvent) {
+      const { keyCode } = e
+      if (keyCode === 13 || keyCode === 39) { // Enter
+        e.preventDefault()
+      }
+    },
+    onHandleKeyUp (e: KeyboardEvent) {
+      const { keyCode } = e
+
+      if (keyCode === 13 || keyCode === 39) { // Enter
+        this.$emit('submit')
+        return
+      } else if (keyCode === 8 && !this.value && this.usernames.length) { // Backspace
+        this.remove(this.usernames[this.usernames.length - 1])
+      }
+
+      // NOTE: this.$refs.input.innerHTML could contains HTML entities
+      this.value = decode(this.$refs.input.innerHTML)
```

Comment:
What is going on here? This is very strange...

- why aren't you using `v-model`?
- if you must do it this way for some reason, it should be clearly commented and explained why `v-model` isn't used
- why add a new dependency instead of [using `textContent`](https://medium.com/@stevriss22/differences-between-innertext-textcontent-innerhtml-23268b2ac7b2)?

Reply:
> why aren't you using v-model?

We can use `v-model` for only `input`, `select`, `textarea` elements. Reference in [here](https://vuejs.org/api/built-in-directives.html#v-model)
> if you must do it this way for some reason, it should be clearly commented and explained why v-model isn't used

Okay
> why add a new dependency instead of using textContent?

Correct. `textContent` works fine. I used `html-entities` library to handle HTML entities as it's not an input field. To be honest, never knew `textContent` before. Thanks for letting me know.